### PR TITLE
Feature/3 modlog

### LINKED ISSUE #3: ModLog

### PR DESCRIPTION
**Pull request by TheJuki**
Closes #3 

Also moved Date Formats to DateUtils.FRONT_END_DATE_FORMAT

Also updated UserController to use getPersonOrThrowUnauthorized(principal); - This was blowing up if the frontend cookie expired.

### CODE REVIEW

**Review by Pdzly — DISMISSED** [review]
Everything looks good! But let jgrim have an eye on it too!

File: src/main/resources/db/migration/V20231216__Alter_moderation_logs.sql
Comment (Pdzly) [inline src/main/resources/db/migration/V20231216__Alter_moderation_logs.sql:1]: Just a question, why not doing it in the initial migration? Yes this should NOT be done IF it was released, but it isnt yet so the db will be cleared on deploy to the test env.

Comment (jgrim) [reply src/main/resources/db/migration/V20231216__Alter_moderation_logs.sql:1]: Please move this into the initial migration.

File: src/main/java/com/sublinks/sublinksapi/utils/DateUtils.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.sublinks.sublinksapi.utils;
+/**
+ * Date Utils
+ */
+public class DateUtils {
+  public static final String FRONT_END_DATE_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSSSSSX";
```
Comment (jgrim) [inline src/main/java/com/sublinks/sublinksapi/utils/DateUtils.java:6]: This format is unique to Lemmy APIs and should be kept within the Lemmy package.

**Review by jgrim — CHANGES_REQUESTED** [review]
Looks great! Please move the migration into the initial migration.

File: src/main/java/com/sublinks/sublinksapi/utils/DateUtils.java (hunk 1 above)
Comment (TheJuki) [reply src/main/java/com/sublinks/sublinksapi/utils/DateUtils.java]: Moved DateUtils

File: src/main/resources/db/migration/V20231216__Alter_moderation_logs.sql
Comment (TheJuki) [reply src/main/resources/db/migration/V20231216__Alter_moderation_logs.sql:1]: Moved columns

File: src/main/resources/db/migration/V20231003__Create_initial_entity_tables.sql [hunk 2]
```diff
@@ -429,14 +430,13 @@ CREATE TABLE `moderation_logs`
...
   `banned`               TINYINT                                   NULL,
   `featured`             TINYINT                                   NULL,
+  `featured_community`   TINYINT                                   NULL,
   `expires`              TIMESTAMP(3)                              NULL,
   `created_at`           TIMESTAMP(3) DEFAULT CURRENT_TIMESTAMP(3) NOT NULL
 ) ENGINE = InnoDB
   DEFAULT CHARSET `utf8mb4`
   COLLATE = 'utf8mb4_unicode_ci';
 
-CREATE UNIQUE INDEX `IDX_MODERATION_LOGS_INSTANCE_ID_ACTION_TYPE_ENTITY_ID` ON `moderation_logs` (`instance_id`, `action_type`, `entity_id`);
```
Comment (jgrim) [inline src/main/resources/db/migration/V20231003__Create_initial_entity_tables.sql:438]: Why are you removing the index?

**Review by jgrim — CHANGES_REQUESTED** [review]
I am curious why the index is being removed.

File: src/main/resources/db/migration/V20231003__Create_initial_entity_tables.sql (hunk 2 above)
Comment (TheJuki) [reply src/main/resources/db/migration/V20231003__Create_initial_entity_tables.sql:438]: We can still create an index, just not unique. Most mod actions can be undone which creates a new modlog for the same action type where one of the TINYINT columns is inverted.

Comment (jgrim) [reply src/main/resources/db/migration/V20231003__Create_initial_entity_tables.sql:438]: Good point, I think we should add some indexes that reflect the query to help improve lookups.

Comment (TheJuki) [reply src/main/resources/db/migration/V20231003__Create_initial_entity_tables.sql:438]: Added indexes and foreign keys

**Review by jgrim — APPROVED** [review]

**Pull request merged**